crates/ruff_linter/src/rules/flake8_builtins/snapshots/ruff_linter__rules__flake8_builtins__tests__A003_A003.py.snap

```diff
@@ -10,6 +10,15 @@ A003.py:17:31: A003 Python builtin is shadowed by method `str` from line 14
 18 |         pass
    |
 
+A003.py:20:34: A003 Python builtin is shadowed by class attribute `id` from line 8
```

Comment:
The reason this is happening is that the self.id is shadowing class attribute `id` which shadows Python builtin ID. In the A003 rule we go over all the references of the binding that shadows the builtin, and in this case self.id is one reference to class attribute id so this warning is triggered.

I was thinking what should be the right thing to do here, we can do either of:

1. Not count self.id as a reference to id. https://michaelgoerz.net/notes/a-note-on-python-class-attributes.html
2. When going over references in the rule exclude the instance attribute references. This would fix this rule but I don't want to leak a wrong data model to ruff and just fix the symptom.